Cartfile

```diff
@@ -3,3 +3,4 @@ binary "https://api.mapbox.com/downloads/v2/carthage/mobile-navigation-native/Ma
 github "mapbox/mapbox-directions-swift" "v2.0.0-beta.3"
 github "mapbox/turf-swift" "v2.0.0-alpha.3"
 github "mapbox/mapbox-events-ios" ~> 0.10.2
+github "pointfreeco/swift-snapshot-testing" ~> 1.8.0
```

Comment:
> > * testAdjacentShields.1.png is only showing generic shields rather than graphical shields. It’s also showing a `/` between the generic shields. Ideally it wouldn’t, but I don’t remember how we currently handle that case.
> 
> I have problems with this one. The issue seems to be in broken links to the shields.

Ok, this is a good one. 
The problem is that `InstructionLabel.update()` clears `imageCache` in too many cases. In other words, you can say that the image cache doesn't work at all with `InstructionLabel`s are used. 

So here it is what happens:
1. Each time any style property is set on `StylableLabel`, it calls the `update()` method. 
1. Update method on `InstructionLabel` calls `imageRepository.resetImageCache(nil)`, which removes all the cache images. Anything cached will be cleared on the next card appearance. 

This regression is introduced in https://github.com/mapbox/mapbox-navigation-ios/pull/2977, which fixed https://github.com/mapbox/mapbox-navigation-ios/issues/2938. 

So we need a more robust cache clear logic. 

Cc @mapbox/navigation-ios I would like to discuss this bug with somebody who is has a deep understanding of how images are cached and styled to work out a plan for the fix.

Reply:
Perhaps the call to `ImageRepository.resetImageCache()` should be moved from `InstructionLabel.update()` to `traitCollectionDidChange(_:)`?

Reply:
Thanks for help. I will deploy a temporary fix and will start looking into the issue separately.